components/header/src/index.js

```diff
@@ -73,4 +73,5 @@ Header.propTypes = {
 export default deprecate(Header, 'please use the Heading component instead');
 
 export { Header as DocumentedHeader };
+// eslint-disable-next-line import/no-cycle
 export { H1, H2, H3, H4, H5, H6 } from './presets';
```

Comment:
the point of this linting warning is that `presets` is importing this file, thus this is a circular dependency

ideally we'd work around this by having the `Header` component in a separate `atom` file, and import that into the `index` file, allowing the `presets` file to import from the `atom` file, thus ensuring there is no circularity

unfortunately this doesn't work with our documentation generation system.  attempting to do that results in empty documentation

another possibility would be to define H1-H6 inside this file, however that too is incompatible with our documentation generation system, as it results in multiple components being defined inside this file

Reply:
please can you open an issue to review this in 0.9?